packages/phrase/src/phrase-api.ts

```diff
@@ -114,40 +114,64 @@ export async function pullAllTranslations(
   return translations;
 }
 
-export async function pushTranslationsByLocale(
-  contents: TranslationsByKey,
-  locale: string,
-  branch: string,
+export async function pushTranslations(
+  translationsByLanguage: TranslationsByLanguage,
+  { devLanguage, branch }: { devLanguage: string; branch: string },
 ) {
   const formData = new FormData();
-  const fileContents = Buffer.from(JSON.stringify(contents));
+  const { csvString, localeMapping, keyIndex, commentIndex, tagColumn } =
+    translationsToCsv(translationsByLanguage, devLanguage);
+  const fileContents = Buffer.from(csvString);
   formData.append('file', fileContents, {
-    contentType: 'application/json',
-    filename: `${locale}.json`,
+    contentType: 'text/csv',
+    filename: `translations.csv`,
   });
 
-  formData.append('file_format', 'json');
-  formData.append('locale_id', locale);
+  formData.append('file_format', 'csv');
   formData.append('branch', branch);
   formData.append('update_translations', 'true');
 
-  log('Starting to upload:', locale, '\n');
+  for (const [locale, index] of Object.entries(localeMapping)) {
+    formData.append(`locale_mapping[${locale}]`, index);
+  }
 
-  const { id } = await callPhrase<{ id: string }>(`uploads`, {
+  formData.append('format_options[key_index]', keyIndex);
+  formData.append('format_options[comment_index]', commentIndex);
+  formData.append('format_options[tag_column]', tagColumn);
+  formData.append('format_options[enable_pluralization]', 'false');
+
+  log('Uploading translations');
+
+  const result = await callPhrase<
+    | {
+        id: string;
+      }
+    | {
+        message: string;
+        errors: Array<unknown>;
+      }
+  >(`uploads`, {
     method: 'POST',
     body: formData,
   });
-  log('Upload ID:', id, '\n');
-  log('Successfully Uploaded:', locale, '\n');
 
-  return { uploadId: id };
+  trace('Upload result:\n', result);
+
+  if ('id' in result) {
+    log('Upload ID:', result.id, '\n');
+    log('Successfully Uploaded\n');
+  } else {
+    log(`Error uploading: ${result?.message}\n`);
```

Comment:
If result is undefined then line 160 will error. If this optional chain is required then I'd add a check to the if statement on line 160

Reply:
There's probably a contract I could reference somewhere, I doubt the response from the phrase calls will ever be `undefined`, but I've added it to the generic just in case.